messaging/repository/room_repository.go

```diff
@@ -44,167 +15,108 @@ func NewRoomRepository(session *gocql.Session) *RoomRepository {
-
-	return err
-
+func (r RoomRepository) AddParticipantToRoom(studentID string, roomID string) error {
+	return r.dbSession.Query(addParticipantToRoom, [1]string{studentID}, roomID).Consistency(gocql.One).Exec()
 }
 
+func (r RoomRepository) DeleteRoom(roomID string) error {
+	return r.dbSession.Query(deleteRoom, roomID).Consistency(gocql.One).Exec()
+}
 
-func (r RoomRepository) GetRoom( roomID string) (*domain.ChatRoom, error) {
+func (r RoomRepository) GetRoom(roomID string) (*domain.ChatRoom, error) {
 	var room domain.ChatRoom
-
 	var studentText  []string
-	var AllStudent []domain.Student
-
-	err := r.dbSession.Query(GetRoom, roomID).Consistency(gocql.One).Scan(&room.RoomID, &room.Admin.ID,&room.Deleted, &room.Name, &studentText);
-
+	var allStudents []domain.Student
 
+	err := r.dbSession.Query(getRoom, roomID).Consistency(gocql.One).Scan(&room.RoomID, &room.Admin.ID,&room.Deleted, &room.Name, &studentText)
 	if err!=nil {
-		return nil,err ;
+		return nil, err
 	}
 
 	for _,ID := range studentText{
 		var student domain.Student
 		student.ID=ID
-		AllStudent = append(AllStudent,student )
-
+		allStudents = append(allStudents,student)
 	}
-	room.Students= AllStudent
-
-
-
-
+	room.Students = allStudents
 
 	return &room,err
-
 }
 
+func (r RoomRepository) RemoveParticipantFromRoom(studentID string, roomID string) error {
+	return r.dbSession.Query(removeParticipantFromRoom, [1]string{studentID}, roomID).Consistency(gocql.One).Exec()
+}
 
-func (r RoomRepository) EditChatRoomParticipants( roomID string, student []domain.Student) error {
-	var studentID []string
-
-	for _,s := range student{
+func (r RoomRepository) SaveRoom(room *domain.ChatRoom) error {
+	var studentIDArr []string
+	for _, student := range room.Students{
+		studentIDArr = append(studentIDArr, student.ID)
+	}
+	return r.dbSession.Query(saveRoom, room.RoomID,  room.Name, room.Admin.ID, studentIDArr).Consistency(gocql.One).Exec()
+}
 
-		studentID = append(studentID,s.ID )
+func (r RoomRepository) AddRoomForParticipant(roomID string, studentID string) error {
+	return r.dbSession.Query(addRoomForParticipant, [1]string{roomID}, studentID).Consistency(gocql.One).Exec()
+}
 
+func (r RoomRepository) AddRoomForParticipants(roomID string, studentIDs []string) error {
+	for _,id := range studentIDs {
+		err := r.dbSession.Query(addRoomForParticipant, [1]string{roomID}, id).Consistency(gocql.One).Exec()
+		if err!=nil {
+			return err
+		}
 	}
-	err := r.dbSession.Query(editChatroomParticipant, studentID, roomID).Consistency(gocql.One).Exec();
-	if err!=nil {
-		return err ;
-	}
-
 	return  nil
 }
 
-func (r RoomRepository) GetRoomsFor( studentID string) (*domain.StudentChatRooms, error) {
+func (r RoomRepository) GetRoomsFor(studentID string) (*domain.StudentChatRooms, error) {
 	var StudentRoom domain.StudentChatRooms
-	var roomsID []string
+	var roomsID []string // used to unmarshall entire set to [0]th entry in string array
```

Comment:
does it unmarshall a set to 1 single string? If so, why do we need an array instead of a string?

Reply:
yes it unmarshalls the set into 1 string entry, it says cannot unmarshall set to string if I use string, but it works when I unmarshall to an array